docs/validator.md

```diff
@@ -0,0 +1,261 @@
+      }
+  }
+  ```
+
+This component is configured to work together with a `gpii.schema.parser` and to further evolve the error messages
+where possible.  See [the parser documentation](parser.md) for details.
+
+# Configuration Options
+
+You are expected to configure one or more schemas which will be read from `options.schemaContents`.  The `schemaKey`
+value passed to `validate` is expected to match one of the keys in `options.schemaContent`.
+
+You can change the behavior of AJV by updating `options.validatorOptions` (check their documentation for the syntax).
+
+# Reusing Content between Schemas
+
+One of the key strengths of JSON Schema is that it allows you to compose a complex schema out of parts taken from
+other schemas.  The validator in this package supports references between JSON Schemas.
+
+Inheritance is still a sticking point at least in v4 of the draft standard.   You cannot safely expect to overlay
+multiple schemas on top of each other.  Best practice for now is to only reuse individual definitions between schemas,
+and to explicitly specify each schema's required properties.
+
+A limitation of this implementation is that it expects external references to use filenames rather than ids,
+as in:
+
+    $ref: "filename.json#/definitions/field"
+
+This is a byproduct of `json-schema-ref-parser`, the library we use to evolve the error output. As this may change, best
+practice is for the `id` in the schema to exactly match the filename.  See the `derived.json` and `base.json` test
+schemas for an example.
+
+# Components
+
+## gpii.schema.validator.ajv
+
+The common grade, designed for use on the client and server-side.
+
+## gpii.schema.validator.ajv.server
+
+A server-side component which additionally populates `schemaContents` for you on startup based on the contents of
+`options.schemaPath`.  Because of limitations in the current AJV `parser`, the schema key must match the filename.
+
+The server-side component will also resolve dependencies from `schemaContents`.  For example, if you have a second
+schema that has a reference to `#schema-file-name.json`, it will resolve to the contents of `schemaContents["schema-file-name.json"]`.
+
+# Functions
+
+## `gpii.schema.validator.ajv.validate(that, key, content)`
+
+* `that`: The component itself.
+* `key {String}`:  The schema key we are validating against.
+* `content {Object}`: The JSON content we are validating.
+* Returns: An `{Object}` describing the errors, or `undefined` if there are no errors.
+
+Transform raw validator output into a more human-readable form that corresponds to the structure of the original JSON
+input.  Typically accessed using the validator component's `validate` invoker and the last two arguments.
+
+
+## `gpii.schema.validator.ajv.sanitizeValidationErrors(that, schemaKey, errors)`
+
+* `that`: The component itself.
+* `schemaKey {String}`: The schema key we are validating against.
+* `errors`: A map of existing error results to be sanitized.
```

Comment:
There was an example of the error format immediately below this. I added a "see below" here.